### PR TITLE
adds refcount to the gc log

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
adds refcount to the gc log

## Why It's Good For The Game
better information for debugging

## Testing
Checked the GC log in game.

## Changelog
NPFC

### CODE REVIEW

File: code/controllers/subsystem/SSgarbage.dm [hunk 1]
```diff
@@ -258,6 +259,7 @@ SUBSYSTEM_DEF(garbage)
 
 	I.hard_deletes++
 	I.hard_delete_time += TICK_DELTA_TO_MS(tick)
+	I.reference_average =  (refcount(I) + I.reference_average) / I.hard_deletes
```
Comment (GDNgit) [inline code/controllers/subsystem/SSgarbage.dm:262]:
```suggestion
	I.reference_average = (refcount(I) + I.reference_average) / I.hard_deletes
```

**Comment by warriorstar-orion** [comment]
Hallelujah

Two quick questions: 

1. Is refcount expensive?
2. Apparently just the act of calling refcount can result in byond deleting the object:

> A return value of 0 can mean one of several things: Either this was the last reference and the object was subsequently deleted after refcount(), or the value doesn't support reference counting.

Is that something we should be concerned about/account for?

**Comment by GDNgit** [comment]
> Hallelujah
> 
> Two quick questions:
> 
> 1. Is refcount expensive?

No, it's actually rather cheap as far as built ins go

> 2. Apparently just the act of calling refcount can result in byond deleting the object:
> 
> > A return value of 0 can mean one of several things: Either this was the last reference and the object was subsequently deleted after refcount(), or the value doesn't support reference counting.
> 
> Is that something we should be concerned about/account for?

They mean the last reference to the object, aka the only remaining object was the argument itself and it would be gc'd by net soon after.

**Comment by warriorstar-orion** [comment]
> Hallelujah
> 
> Two quick questions:
> 
> 1. Is refcount expensive?
> 2. Apparently just the act of calling refcount can result in byond deleting the object:
> 
> > A return value of 0 can mean one of several things: Either this was the last reference and the object was subsequently deleted after refcount(), or the value doesn't support reference counting.
> 
> Is that something we should be concerned about/account for?

Discussed offline, refcount calls should be cheap because they're just a C primitive internally

**Review by warriorstar-orion — APPROVED** [review]
LGTM, would like another pair of eyes on it though

**Review by lewcc — APPROVED** [review]
LGTM

**Pull request merged**